src/common/com/intellij/plugins/haxe/model/IHaxeEnumModel.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 Ilya Malanin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public interface IHaxeEnumModel {
```

Comment:
Ah, that was a compromise. I've already introduced `HaxeEnumModel`, and there was a situation when a common interface became needed for `HaxeEnumModel` and `HaxeAbstractEnumModel`. Since the name was already taken, I haven't invent anything better than to name it explicitly with I prefix.

Reply:
I'm not sure about convention, may be you could offer a better solution?

Reply:
Why don't you use the HaxeEnumModel name for the interface and rename the current class HaxeEnumModelImpl?  They're both being introduced in this PR anyway.